### ISSUE TITLE
[Feature request] Support for non-input types + variables

### ISSUE THREAD

**Issue by robertherber** [Code-Hex/graphql-codegen-typescript-validation-schema#403]
Thanks @Code-Hex for creating this plugin! :)

We're doing end-to-end typesafety all the way from MongoDB to the client with our GraphQL schema as a base. Currently we're doing it TypeScript-only, but looking to improve on the DX as well as strictness by using zod both for client-side validation as well as using it to enforce [MongoDB Schema Validation](https://www.mongodb.com/docs/manual/core/schema-validation/).

For this to work well we need to generate zod definitions for:
- Non-input types as well - we need to cover all GraphQL types as a base for MongoDB Schema Validation ([will probably use this or something similar](https://github.com/marcesengel/zod-to-mongodb-schema))
- I noticed zod definitions are not generated for variables directly specified in mutations. I think it'd help a lot to get these as well - I think it makes most sense to break out input types only when they reach a certain level of complexity.

Maybe this is already possible in some way and I've missed how to configure it - but I feel both things would be very helpful to utilize this plugin to it's full potential! :)

**Comment by Code-Hex** [comment]
@robertherber Thanks!
Non-input types are available with `withObject`. Please read README more details.

However, I don't support variable for now. Need implementation.

**Comment by Code-Hex** [comment]
Could you tell me example use-case and the code?

I want to reference as real world examples for implementation.

**Comment by robertherber** [comment]
When using `withObjectType` I end up with this cryptic error message:
<img width="521" alt="Screenshot 2023-06-26 at 08 14 41" src="https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/assets/1467411/0b0b2150-c480-43bf-b86b-4ba8fe4f8b94">

And my Codegen looks like this:
```yml
overwrite: true
schema: ***
generates:
  src/__generated__/schema.ts:
    plugins:
      - typescript
      - typescript-validation-schema
    config:
      useTypeImports: true
      schema: zod
      nonOptionalTypename: true
      immutableTypes: true
      withObjectType: true # <- without this it works and I get the input types
      directives:
        minLength: min
        # Replace $1 with specified `startsWith` argument value of the constraint directive
        startsWith: [ matches, /^$1/ ]
        format:
          email: email
      scalars:
        DateTime: string
        ObjectId: string
      namingConvention:
        enumValues: keep
```

As for inputs I would like to be able to use mutations like this:
```GraphQL
mutation Login(email: String!, password: String!){
  token
}

// the option now - as far as I understand it is making it extra verbose like this:
input LoginInput {
  email: String!, 
  password: String!
}

mutation Login(loginArgs: LoginInput!){
  token
}
```

**Comment by Code-Hex** [comment]
I'm sorry for being late.
Just stating the facts:
- The code you provided is a document, not a schema.
  - See: https://stackoverflow.com/questions/59417171/whats-difference-between-schema-and-documents-in-graphql
- This tool supports schemas but not documents.

**Comment by Code-Hex** [comment]
@robertherber I'm sorry for being late.
Just stating the facts:
- The code you provided is a document, not a schema.
  - See: https://stackoverflow.com/questions/59417171/whats-difference-between-schema-and-documents-in-graphql
- This tool supports schemas but not documents.

**Comment by robertherber** [comment]
Sorry for making it confusing by not providing a syntactically correct schema, I've fixed the example now.

The point is still that I'd like to be able to use arguments directly sent to a mutation and not rely on creating intermediate Input types when it's not necessary.

**Comment by Code-Hex** [comment]
@robertherber I'm not sure what are you expected typescript code generation if I can implement to generate via variables.

As I mentioned earlier, variables are document syntax, not schema.
https://spec.graphql.org/June2018/#sec-Language.Variables

**Comment by robertherber** [comment]
I’m not talking about documents but the schema itself which I hope is clear after I updated the example 🙂

**Comment by Code-Hex** [comment]
@robertherber I meant What do you want Zoe schema for variable If you defined this document?
```
mutation Login(email: String!, password: String!): String!
```

**Comment by Code-Hex** [comment]
@robertherber I created PR for variables but only for schema.
https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/pull/453

**Comment by Code-Hex** [comment]
I just published this as 0.12.0
https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/tree/v0.12.0

**Comment by SuttonKyle** [comment]
This seems to have had a regression recently -- the variables are no longer generated and it looks like the block that was used to add them has been removed.
